src/main/java/uk/gov/hmcts/reform/sscs/services/CaseLoaderService.java

```diff
@@ -46,50 +52,120 @@
                         file.getName()));
                 }
+
                 processDelta(file);
+                log.info(logPrefixWithFile + " Summary: \nStart: {}\nEnd: {}\nSize: {}\nRecords: {}\nTime: {}",
+                    fileMetrics.getStartTime().format(DateTimeFormatter.ISO_DATE_TIME),
+                    fileMetrics.getEndTime().format(DateTimeFormatter.ISO_DATE_TIME),
+                    fileMetrics.getFileSize(),
+                    fileMetrics.getRecordCount(),
+                    fileMetrics.getRunTime()
+                );
+
+                metrics.merge(fileMetrics);
+
                 sftpSshService.move(file, true);
                 sftpSshService.move(latestRef, true);
             } else {
                 latestRef = file;
                 try {
                     refDataFactory.extract(sftpSshService.readExtractFile(file));
                 } catch (XMLStreamException e) {
-                    throw new TransformException("Error processing reference file", e);
+                    throw new TransformException(logPrefixWithFile + " Error processing reference file", e);
                 }
             }
         }
+
+        metrics.setEndTime();
+
+        log.info(logPrefix + " End Summary: \nStart: {}\nEnd: {}\nSize: {}\nRecords: {}\n"
+            + "Total Time: {}\nFiles Processed: {}",
+            metrics.getStartTime().format(DateTimeFormatter.ISO_DATE_TIME),
+            metrics.getEndTime().format(DateTimeFormatter.ISO_DATE_TIME),
+            metrics.getFileSize(),
+            metrics.getRecordCount(),
+            metrics.getRunTime(),
+            metrics.getFileName()
+        );
     }
 
     private void processDelta(Gaps2File file) {
+        fileMetrics = new CaseLoaderMetrics();
+        fileMetrics.setStartTime();
+        fileMetrics.setFileName(file.getName());
+        fileMetrics.setFileSize(transformService.getLastStreamLength());
+
         List<SscsCaseData> cases = transformService.transform(sftpSshService.readExtractFile(file));
-        log.info("*** case-loader *** file transformed to {} Cases successfully", cases.size());
+
+        fileMetrics.setRecordCount(cases.size());
+
+        log.info(logPrefixWithFile + " file transformed to {} Cases successfully", cases.size());
         int counter = 0;
         IdamTokens idamTokens = idamService.getIdamTokens();
         for (SscsCaseData caseData : cases) {
             if (!caseData.getAppeal().getBenefitType().getCode().equals("ERR")) {
                 idamTokens.setServiceAuthorization(idamService.generateServiceAuthorization());
+                
                 if (counter == numberOfProcessedCasesToRefreshTokens) {
                     idamTokens.setIdamOauth2Token(idamService.getIdamOauth2Token());
                     log.info("*** case-loader *** renew idam token successfully");
                     counter = 0;
                 }
+                SscsCaseDetails sscsCaseDetails;
```

Comment:
Be careful here, this logic was already in place, you are duplicating it. You also changed junit and integration tests as a result of this duplication.